src/Altinn.App.Core/Internal/Process/ProcessEventHandlingDelegator.cs

```diff
@@ -109,6 +73,63 @@ IEnumerable<IProcessTask> processTasks
             }
         }
 
+        private async Task HandleTaskEvent(
+            Instance instance,
+            InstanceEventType eventType,
+            ProcessElementInfo currentTask,
+            Dictionary<string, string>? prefill
+        )
+        {
+            string taskId = currentTask.ElementId;
+            string altinnTaskType = currentTask.AltinnTaskType;
+
+            switch (eventType)
+            {
+                case InstanceEventType.process_StartTask:
```

Comment:
Might be better to do command pattern instead of these switches, but that's a challenge for later.